src/gazebo_mavlink_interface.cpp

```diff
@@ -554,18 +554,16 @@ void GazeboMavlinkInterface::Load(physics::ModelPtr _model, sdf::ElementPtr _sdf
 // This gets called by the world update start event.
 void GazeboMavlinkInterface::OnUpdate(const common::UpdateInfo&  /*_info*/) {
 
-  std::unique_lock<std::mutex> lock(last_imu_message_mutex_);
+  std::unique_lock<std::mutex> lock(imu_received_mutex_);
 
-  if (previous_imu_seq_ > 0) {
-    while (previous_imu_seq_ == last_imu_message_.seq() && IsRunning()) {
-      last_imu_message_cond_.wait_for(lock, std::chrono::milliseconds(10));
+  if (imu_received_once_) {
+    while (!imu_received_ && IsRunning()) {
+      imu_received_cond_.wait_for(lock, std::chrono::milliseconds(10));
     }
   }
-
-  previous_imu_seq_ = last_imu_message_.seq();
-
+  imu_received_ = false;
   // Always run at 250 Hz. At 500 Hz, the skip factor should be 2, at 1000 Hz 4.
-  if (!(previous_imu_seq_ % update_skip_factor_ == 0)) {
+  if (!(update_counter_++ % update_skip_factor_ == 0)) {
```

Comment:
I think `update_counter_++` should only increment on the primary imu

Reply:
update_counter_ is already independent on the number of IMUs. the counter is only incremented in the OnUpdate function, which should only be called once every world iteration and not dependent on the imuCallback.

Reply:
Ah, right thanks